### PR TITLE
added export method :as_css

### PR DESCRIPTION
**Pull request by krtschmr**
As written in https://github.com/whomwah/rqrcode/issues/77 it supports to export as pure HTML so it can be styled with css. As a bonus it accepts a path to a logo which can be overlayed.

`.as_css` takes arguments as a hash for `size`, `logo`, `logo_size`, `id`, `class`

The logo should always be centered. 

Example CSS is provided, Readme updated, specs added and all tests passing.

image

### CODE REVIEW

**Comment by krtschmr** [comment]
@bjornblomqvist how you like this?

**Comment by krtschmr** [comment]
@whomwah can we get this reviewed and merged?

**Comment by whomwah** [comment]
Thank you for the PR. I'll get round to reviewing this as soon as time allows. I only work on the project a couple of hours a month.

**Comment by shaicoleman** [comment]
Would be better to use CSS classes instead of IDs, to allow multiple QR codes on one page if needed.

**Comment by krtschmr** [comment]
good idea, @shaicoleman . i can implement that on the weekend

**Comment by whomwah** [comment]
Hi all, sorry for the delaying in getting back. Busy life.

I'm in the process of updating the README for this project for future direction, but the long and short of it is that the current `as_png`, `as_css`, `as_svg`, `as_ansi` etc etc renderings of a QR Code will all be moved into their own gems so they can be managed independently by people in their own repos -- ideally -- who are interested in this kind of thing. This will leave me to look after `rqrcode_core` gem which I do have time for.

So for example if you only required a `png` rendering of a QR Code, you could simply use the gem `rqrcode_png`. This eventually means that the `rqrcode` gem will just become a lightweight gem that pulls in all the various rendering and TBH will likely be depreciated.

The motivation behind all this is because the rendering side of this gem takes up the most time. Everyone wants a slightly different version of a QR Code and it's impossible to support everyone. The easiest way is to empower people to create their own which they can manage.

Hope this is clear, and like I said I will add a more thorough explanation to the README and a roadmap on when this will be done. The work shouldn't impact any current users of the gem.

**Comment by stephanpavlovic** [comment]
Any progress on this or the separated gems for rendering. Would love to use the css part...

**Comment by gresbtim** [comment]
@stephanpavlovic i never had time to implement that. however you can just extract the CSS methods out of my pull and that's working perfect. i used it in multiple projects

File: lib/rqrcode/export/css.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
+module RQRCode
+  module Export
+    module CSS
+
+      def as_css(options={})
+        id = options.fetch(:id, "qrcode")
+        clazz = options.fetch(:id, "qrcode-container")
+        base_size = options.fetch(:size, 200)
+        @@qr_logo = options.fetch(:logo, nil)
```
Comment (korun) [inline lib/rqrcode/export/css.rb:9]: Why do we need to class variables here..?
It is [bad style](https://github.com/rubocop-hq/ruby-style-guide#no-class-vars), and and it looks like it’s not threadsafe.

Comment (gresbtim) [reply lib/rqrcode/export/css.rb:9]: how would you like to change it?

Comment (korun) [reply lib/rqrcode/export/css.rb:9]: I think we can just use instance variables: `@qr_logo` instead of `@@qr_logo`

**Comment by mjy** [comment]
@whomwah updating the README, even with a brief note, even if not complete, would help re the seperate gemset ideas and help to avoid unecessary PRs (like my recent ones) perhaps)?

**Comment by whomwah** [comment]
@mjy fair point. I will update the `Contributing` section.

**Pull request closed without merging**